restcomm/restcomm.testsuite/src/test/java/org/mobicents/servlet/restcomm/http/RestcommGeolocationsTool.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2016, Telestax Inc and individual contributors
+ * by the @authors tag.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+        
+package org.mobicents.servlet.restcomm.http;
+
+import java.io.IOException;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.WebResource;
+import com.sun.jersey.api.client.filter.HTTPBasicAuthFilter;
+
+/**
+ * @author fernando.mendioroz@telestax.com (Fernando Mendioroz)
+ *
+ */
+public class RestcommGeolocationsTool {
+
+    private static RestcommGeolocationsTool instance;
+    private static String geolocationsUrl;
+
+    public static RestcommGeolocationsTool getInstance() {
```

Comment:
In this case, I honestly think you should take the effort (which should be minimum) and prevent the Singleton, its bad for performance and testability. In addition, im not sure that static geolocationsUrl is properly proptected against concurrent access (have you tried using this tool concurrently in your JUnit test???). If this is not exactly under geoloc Scope, file an issue against it...